staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -808,6 +808,31 @@ func (c *Cacher) dispatchEvents() {
 	}
 }
 
+func (c *Cacher) setCachingObjects(event *watchCacheEvent) {
+	if object, err := newCachingObject(event.Object); err == nil {
+		event.Object = object
+	} else {
+		klog.Errorf("couldn't create cachingObject from: %#v", event.Object)
+	}
+
+	// Since newCachingObject performs DeepCopy on underlying object (which
+	// results in non-negligible number of memory allocations), for now we
+	// switch on caching only for events that can only be DELETE events.
```

Comment:
> we switch on caching only for events that can only be DELETE events

I don't understand this... is this saying we only want to cache delete events? Doesn't that severely limit the performance benefit?